### PR TITLE
docs: Update comments with typescript examples to better utilize TSDoc syntax

### PR DESCRIPTION
**Pull request by Josmithr**
Namely:
- Puts code blocks under `@example` or `@remarks` as content-appropriate
- Adds typescript language specification to example code blocks
- Fixes a couple typos and cleans up some formatting

### CODE REVIEW

File: packages/runtime/datastore-definitions/src/jsonable.ts [hunk 1]
```diff
@@ -26,6 +22,12 @@
  *  - Non-finite numbers (`NaN`, `+/-Infinity`) are also coerced to `null`.
  *
  * Also, `Jsonable<T>` does not prevent the construction of circular references.
+ *
+ * @example
+ * Typical usage:
+ * ```typescript
+ *      function foo<T>(value: Jsonable<T>) { ... }
```
Comment (tylerbutler) [inline packages/runtime/datastore-definitions/src/jsonable.ts:29]: nit: why the indentation?

**Review by tylerbutler — COMMENTED** [review]
🔤

**Review by tylerbutler — APPROVED** [review]

**Review by ssimic2 — APPROVED** [review]
We could possibly finally doc some of the best practices (ex, keep description short and use remarks for examples, further details etc.) of course it’s best when your code is actual reference for “best practices”, but I’m not sure how far off we are there :)

File: packages/runtime/datastore-definitions/src/jsonable.ts (hunk 1 above)
Comment (Josmithr) [reply packages/runtime/datastore-definitions/src/jsonable.ts:29]: Good catch! There shouldn't be any.
```suggestion
 * function foo<T>(value: Jsonable<T>) { ... }
```

**Comment by Josmithr** [comment]
> We could possibly finally doc some of the best practices (ex, keep description short and use remarks for examples, further details etc.) of course it’s best when your code is actual reference for “best practices”, but I’m not sure how far off we are there :)

I have some work items scheduled for this month to write up some of these docs 🙂

**Comment by Josmithr** [comment]
Namely:
- Puts code blocks under `@example` or `@remarks` as content-appropriate
- Adds typescript language specification to example code blocks
- Fixes a couple typos and cleans up some formatting

**Pull request merged**